infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -388,19 +392,46 @@ public void onClose(KubernetesClientException cause) {
   }
 
   private void handleStartingPodStatus(CompletableFuture<Void> podRunningFuture, Pod pod) {
-    if (POD_STATUS_PHASE_RUNNING.equals(pod.getStatus().getPhase())) {
-      podRunningFuture.complete(null);
+    PodStatus status = pod.getStatus();
+    String podPhase = status.getPhase();
+    if (POD_STATUS_PHASE_RUNNING.equals(podPhase)) {
+      // check that all the containers are ready...
+      Map<String, String> terminatedContainers = new HashMap<>();
+
+      for (ContainerStatus cs : status.getContainerStatuses()) {
```

Comment:
imho stream would look better here.

Reply:
I need to access data both from the `cs` variable as well as from its optionally null field `terminated` so in the value mapper of the `toMap` collector I would have to have the same `if` statement I have now in the loop. E.g. no `map` to `terminated` with `null filter` is unfortunately possible, making the stream IMHO unnecessary - I like streams if I can do some processing with them but here it is not the case unfortunately.